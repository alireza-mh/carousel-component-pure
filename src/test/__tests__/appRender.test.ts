import CarouselContainer from "src/components/carouselContainer";

it("renders without crashing", () => {
  document.body.innerHTML = `<div class="root"></div>`;
  const containerNumber = new CarouselContainer({ mountSelector: ".root" });
  const containerNumber2 = new CarouselContainer({ mountSelector: ".root", resetBtn: true });
});

it('renders correctly', () => {
  document.body.innerHTML = `<div class="root"></div>`;
  const containerNumber = new CarouselContainer({ mountSelector: ".root", name: 'test1' });
  const containerNumber2 = new CarouselContainer({ mountSelector: ".root", name: 'test2', resetBtn: true });
  expect(document.querySelector('.root')).toMatchSnapshot();
});
