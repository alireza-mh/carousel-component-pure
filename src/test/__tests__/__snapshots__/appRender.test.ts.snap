// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="root"
>
  
            
  <div
    class="container-carousel"
    id="test1"
  >
    
              
    <form
      action="/"
    >
      
                
      <input
        class="container-input"
        placeholder="insert image link"
        required=""
        type="text"
      />
      
                
      <button
        class="container-btn"
        type="submit"
      >
        Submit
      </button>
      
                
              
    </form>
    
            
    <div
      class="owl-carousel"
      id="test1-carousel"
    >
      
            
    </div>
    
            
  </div>
  
          
  
            
  <div
    class="container-carousel"
    id="test2"
  >
    
              
    <form
      action="/"
    >
      
                
      <input
        class="container-input"
        placeholder="insert image link"
        required=""
        type="text"
      />
      
                
      <button
        class="container-btn"
        type="submit"
      >
        Submit
      </button>
      
                
      <button
        class="container-btn-reset"
        type="reset"
      >
        Reset
      </button>
      
              
    </form>
    
            
    <div
      class="owl-carousel"
      id="test2-carousel"
    >
      
            
    </div>
    
            
  </div>
  
          
</div>
`;
